Question 1: Download and load the Ames CSV file. Read the file into a Pandas DataFrame and display the first five rows.  

In [2]:
import pandas as pd
df =pd.read_csv(r'C:\Users\bsila\Dexian_Bootcamp\code\DataPyLabExercise\DA_labs\AmesHousing.csv\AmesHousing.csv')
df.head()

Order        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street  \
0      1  526301100           20        RL         141.0     31770   Pave   
1      2  526350040           20        RH          80.0     11622   Pave   
2      3  526351010           20        RL          81.0     14267   Pave   
3      4  526353030           20        RL          93.0     11160   Pave   
4      5  527105010           60        RL          74.0     13830   Pave   

  Alley Lot Shape Land Contour  ... Pool Area Pool QC  Fence Misc Feature  \
0   NaN       IR1          Lvl  ...         0     NaN    NaN          NaN   
1   NaN       Reg          Lvl  ...         0     NaN  MnPrv          NaN   
2   NaN       IR1          Lvl  ...         0     NaN    NaN         Gar2   
3   NaN       Reg          Lvl  ...         0     NaN    NaN          NaN   
4   NaN       IR1          Lvl  ...         0     NaN  MnPrv          NaN   

  Misc Val Mo Sold Yr Sold Sale Type  Sale Condition  SalePrice  
0        0       5    2010       WD           Normal     215000  
1        0       6    2010       WD           Normal     105000  
2    12500       6    2010       WD           Normal     172000  
3        0       4    2010       WD           Normal     244000  
4        0       3    2010       WD           Normal     189900  

[5 rows x 82 columns]

Question 2:  Print the highest and lowest priced homes in Ames.

In [6]:
highest_price = df['SalePrice'].max()
print(f"Highest price: ${highest_price}")
Lowest_price= df['SalePrice'].min()
print(f"Lowest price: ${Lowest_price}")

Highest price: $755000
Lowest price: $12789


Question 3:  How many categories does lot size have? 

In [157]:
df.groupby(['Lot Shape']).count()

Order   PID  MS SubClass  MS Zoning  Lot Frontage  Lot Area  \
Lot Shape                                                                
IR1          979   979          979        979           654       979   
IR2           76    76           76         76            48        76   
IR3           16    16           16         16            11        16   
Reg         1859  1859         1859       1859          1727      1859   

           Street  Alley  Land Contour  Utilities  ...  Pool QC  Fence  \
Lot Shape                                          ...                   
IR1           979     27           979        979  ...        5    163   
IR2            76      4            76         76  ...        0      7   
IR3            16      1            16         16  ...        1      1   
Reg          1859    166          1859       1859  ...        7    401   

           Misc Feature  Misc Val  Mo Sold  Yr Sold  Sale Type  \
Lot Shape                                                        
IR1                  37       979      979      979        979   
IR2                   4        76       76       76         76   
IR3                   0        16       16       16         16   
Reg                  65      1859     1859     1859       1859   

           Sale Condition  SalePrice  SalePrice_modified  
Lot Shape                                                 
IR1                   979        979                 979  
IR2                    76         76                  76  
IR3                    16         16                  16  
Reg                  1859       1859                1859  

[4 rows x 82 columns]

Question 4:  What is the range of years for the house listings in the data? 

In [26]:
df[['Year Built', 'Yr Sold', 'Year Remod/Add']].max()

Year Built        2010
Yr Sold           2010
Year Remod/Add    2010
dtype: int64

In [27]:
df[['Year Built', 'Yr Sold', 'Year Remod/Add']].min()

Year Built        1872
Yr Sold           2006
Year Remod/Add    1950
dtype: int64

The range of years for the house listings data is between 1872 and 2010.

Question 5:  What is the total number of all homes sold in Ames for all years? 

In [164]:
df['Yr Sold'].count()

2930

Question 6:  How many nulls or NA are there in the Lot Frontage column? 

In [41]:
df.isna().sum()

Order               0
PID                 0
MS SubClass         0
MS Zoning           0
Lot Frontage      490
                 ... 
Mo Sold             0
Yr Sold             0
Sale Type           0
Sale Condition      0
SalePrice           0
Length: 82, dtype: int64

There are 490 nulls or NA in the Lot Frontage Column

Question 7:  How many items have a value in Alley and what is the total number of  NA in the column? 

In [42]:
df['Alley'].count()

198

There are 198 items that has values in Alley and there are 2732 (2930-198) NA values in the column.

Question 8:  What are all the homes in Gilbert with Yr Sold being 2009 and 2010?

Question 9:  What is the count of homes sold in Gilbert with Yr Sold being 2010? 

In [150]:
df.value_counts('Yr Sold')[2010]

341

In [151]:
df.value_counts('Neighborhood')['Gilbert'] 

165

In [165]:
# Count of homes in Gilbert with Yr Sold being  2010
df.value_counts('Neighborhood')['Gilbert'] & df.value_counts('Yr Sold')[2010] 

5

Question 10:  Return the value of the third row, and fifth column.

In [82]:
df.loc[3, 'Lot Frontage']  #3rd row and fifth column

93.0

Question 11:  Remake a new column from the SalePrice that includes data formatted with the $, and commas for the thousands/millions digits. 

In [85]:
df['SalePrice_modified'] = df['SalePrice'].apply(lambda x: '${:,}'.format(x))
df

Order        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street  \
0         1  526301100           20        RL         141.0     31770   Pave   
1         2  526350040           20        RH          80.0     11622   Pave   
2         3  526351010           20        RL          81.0     14267   Pave   
3         4  526353030           20        RL          93.0     11160   Pave   
4         5  527105010           60        RL          74.0     13830   Pave   
...     ...        ...          ...       ...           ...       ...    ...   
2925   2926  923275080           80        RL          37.0      7937   Pave   
2926   2927  923276100           20        RL           NaN      8885   Pave   
2927   2928  923400125           85        RL          62.0     10441   Pave   
2928   2929  924100070           20        RL          77.0     10010   Pave   
2929   2930  924151050           60        RL          74.0      9627   Pave   

     Alley Lot Shape Land Contour  ... Pool QC  Fence Misc Feature Misc Val  \
0      NaN       IR1          Lvl  ...     NaN    NaN          NaN        0   
1      NaN       Reg          Lvl  ...     NaN  MnPrv          NaN        0   
2      NaN       IR1          Lvl  ...     NaN    NaN         Gar2    12500   
3      NaN       Reg          Lvl  ...     NaN    NaN          NaN        0   
4      NaN       IR1          Lvl  ...     NaN  MnPrv          NaN        0   
...    ...       ...          ...  ...     ...    ...          ...      ...   
2925   NaN       IR1          Lvl  ...     NaN  GdPrv          NaN        0   
2926   NaN       IR1          Low  ...     NaN  MnPrv          NaN        0   
2927   NaN       Reg          Lvl  ...     NaN  MnPrv         Shed      700   
2928   NaN       Reg          Lvl  ...     NaN    NaN          NaN        0   
2929   NaN       Reg          Lvl  ...     NaN    NaN          NaN        0   

     Mo Sold Yr Sold Sale Type Sale Condition  SalePrice  SalePrice_modified  
0          5    2010       WD          Normal     215000            $215,000  
1          6    2010       WD          Normal     105000            $105,000  
2          6    2010       WD          Normal     172000            $172,000  
3          4    2010       WD          Normal     244000            $244,000  
4          3    2010       WD          Normal     189900            $189,900  
...      ...     ...       ...            ...        ...                 ...  
2925       3    2006       WD          Normal     142500            $142,500  
2926       6    2006       WD          Normal     131000            $131,000  
2927       7    2006       WD          Normal     132000            $132,000  
2928       4    2006       WD          Normal     170000            $170,000  
2929      11    2006       WD          Normal     188000            $188,000  

[2930 rows x 83 columns]

Question 12:   Output a dataframe with only SalePrice, Neighborhood, and Year Built.  Save this file in your git repository as “minimal_ames_housing.csv”,  Make sure you include the formatted SalePrice from the previous question.

In [116]:
df2 = df[['Neighborhood', 'Year Built', 'SalePrice', 'SalePrice_modified']]
df2

Neighborhood  Year Built  SalePrice SalePrice_modified
0           NAmes        1960     215000           $215,000
1           NAmes        1961     105000           $105,000
2           NAmes        1958     172000           $172,000
3           NAmes        1968     244000           $244,000
4         Gilbert        1997     189900           $189,900
...           ...         ...        ...                ...
2925      Mitchel        1984     142500           $142,500
2926      Mitchel        1983     131000           $131,000
2927      Mitchel        1992     132000           $132,000
2928      Mitchel        1974     170000           $170,000
2929      Mitchel        1993     188000           $188,000

[2930 rows x 4 columns]

In [117]:
df2.to_csv("minimal_ames_housing.csv", index=False)

In [118]:
import os  
os.makedirs(r'C:\Users\bsila\Dexian_Bootcamp\code\DataPyLabExercise\DA_labs\AmesHousing.csv', exist_ok=True)  
df2.to_csv(r'C:\Users\bsila\Dexian_Bootcamp\code\DataPyLabExercise\DA_labs\AmesHousing.csv\minimal_ames_housing.csv')